2010-05-29  Michael Tautschnig  <mt@debian.org>

	* setup-storage/Init.pm: Properly handle /dev/cciss disk devices, even without
		partitions (thanks Steffen Grunewald <steffen.grunewald@aei.mpg.de> and
		Mathieu Alorent <malorent@echo.fr>).
Index: trunk/lib/setup-storage/Init.pm
===================================================================
--- trunk.orig/lib/setup-storage/Init.pm
+++ trunk/lib/setup-storage/Init.pm	
@@ -183,10 +183,10 @@
     return (1, "/dev/$1", $2);
   }
   elsif ($dev =~
-    m{^/dev/(cciss/c\dd\d|ida/c\dd\d|rd/c\dd\d|ataraid/d\d|etherd/e\d+\.\d+)p(\d+)?$})
+    m{^/dev/(cciss/c\dd\d|ida/c\dd\d|rd/c\dd\d|ataraid/d\d|etherd/e\d+\.\d+)(p(\d+))?$})
   {
     defined($2) or return (1, "/dev/$1", -1);
-    return (1, "/dev/$1", $2);
+    return (1, "/dev/$1", $3);
   }
   # TODO: handle /dev/mapper devices
   return (0, "", -2);
